Automattic/jetpack
# Site Settings API: Expose `show_context` and `show_date` options

Fixes https://github.com/Automattic/wp-calypso/issues/71593

It is necessary to expose those options to implement the **Related posts** setting in the new **Reading Settings** page.

## Proposed changes:

* Extend site settings API to expose `show_context` and `show_date` related posts options for both reading (`GET /sites/%s/settings`) and writing (`POST /sites/%s/settings`)

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Apply the PR to your sandbox
* Sandbox `public-api.wordpress.com`
* Go to wpcom API developer console

#### Testing `POST /sites/%s/settings/`
 * Select `WP.COM API` - `v1.1` - `POST` - `/sites/$site/settings` where `$site` can be your test site's domain or site ID
 * Set `show_context` and/or `show_date` body args to either `true` or `false`
 * Execute the call
 * Validate the response
 * You may double check the values were persisted by executing the `GET /sites/%s/settings/`. See below for instructions.

#### Testing `GET /sites/%s/settings/`
 * Select `WP.COM API` - `v1.1` - `GET` - `/sites/$site/settings` where `$site` can be your test site's domain or site ID
 * Execute the call
 * Validate the response. Ensure the `jetpack_relatedposts_show_context` and the `jetpack_relatedposts_show_date` are returned with the correct values
